lib/index.js

```diff
@@ -853,6 +855,81 @@ GA.prototype.promotionClickedEnhanced = function(track) {
 };
 
 /**
+ * Product List Viewed - Enhanced Ecommerce (Mapped to Product Impression)
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListViewedEnhanced = function(track) {
+  var props = track.properties(); 
+  var products = track.products();
+  this.loadEnhancedEcommerce(track);
+  each(products, function(product) {
+    // If we don't have an ID/SKU or name, return - GA will reject the impression.
+    if (!(product.product_id || product.sku) && !product.name) return;
```

Comment:
can you construct a new Track so we can use facade functions to pull these properties where it can? For example,

```
var item = new Track({ properties: product });
if (!item.productId() || !item.sku()) && !item.name()) return;

.. same below
```

Using facade functions can be helpful because it does bunch of extra lookup stuff sometimes or does case insensitive lookups.